### PR TITLE
Sentry changes

### PR DESCRIPTION
**Pull request by shahthepro**
- Logs all errors in admin pages by default
- Allows the shop admin to enable/disable logging errors for users

### CODE REVIEW

File: shop/src/components/SentryErrorBoundary.js [hunk 1]
```diff
@@ -19,33 +21,58 @@ class ErrorBoundary extends React.Component {
...
+        return null
+      }
+    }
 
-    const pathname = new URL(event.request.url)
-    const isAdminPage = /^#\/(super-)?admin(\/.*)?$/.test(pathname.hash)
-    if (!isAdminPage) {
-      // Ask permission from the user, if it is not in admin page
-      setErrorEvent(event)
+    if (process.env.NODE_ENV === 'development') {
```
Comment (franckc) [inline shop/src/components/SentryErrorBoundary.js:53]: When running on local is NODE_ENV always set to 'development"? Even like in watch mode? Just asking to be sure.

**Comment by franckc** [comment]
Sorry for all the back-end-forth on this... Here is the latest based on the discord discussion on discord with @gullible1 and @matthewliu: For the beta launch they suggest the following logic:
 - Sentry enabled by default for all admin pages for all merchants.
 - Sentry enabled by default on all shops pages for all buyers except on the Brave store where it should be disabled.
We'll revisit this logic for the public launch.

File: backend/routes/shops.js [hunk 2]
```diff
@@ -806,20 +806,25 @@ module.exports = function (router) {
...
-        const network = await Network.findOne({ where: { active: true } })
-        const netConfig = getConfig(network.config)
-        const { secret } = await registerStripeWebhooks(
-          req.body,
-          existingConfig,
-          netConfig.backendUrl || existingConfig.publicUrl
-        )
-        additionalOpts.stripeWebhookSecret = secret
-      } else if (req.body.stripeWebhookSecret) {
+      if (process.env.NODE_ENV !== 'development') {
```
[the remark this answers is not in the document]
Comment (shahthepro) [reply backend/routes/shops.js]: > Also why disabling it exactly? What if we want to test Stripe payment flow on local?

Stripe webhook cannot be registered with localhost. For testing webhooks on local, you have to use stripe CLI and enter the webhook secret manually.

File: shop/src/components/SentryErrorBoundary.js (hunk 1 above)
Comment (shahthepro) [reply shop/src/components/SentryErrorBoundary.js]: Thanks. Changing this to  `!== production`

**Pull request merged**